File: src/main/java/com/github/kindrat/cassandra/client/util/UIUtil.java
package com.github.kindrat.cassandra.client.util;

import com.sun.javafx.tk.FontLoader;
import com.sun.javafx.tk.Toolkit;
import javafx.scene.Node;
import javafx.scene.control.Button;
import javafx.scene.layout.AnchorPane;
import javafx.scene.layout.Region;
import javafx.scene.text.Font;
import lombok.experimental.UtilityClass;
import org.apache.commons.lang3.StringUtils;

import static javafx.scene.layout.Region.USE_PREF_SIZE;

@UtilityClass
public class UIUtil {
    public static void fillParent(Node node) {
        AnchorPane.setLeftAnchor(node, 0.);
        AnchorPane.setRightAnchor(node, 0.);
        AnchorPane.setTopAnchor(node, 0.);
        AnchorPane.setBottomAnchor(node, 0.);
    }

    public static String[] parseWords(String rawString) {
        return StringUtils.split(StringUtils.defaultString(rawString, ""));
    }

    public static void disable(Node... nodes) {
        for (Node node : nodes) {
            node.setDisable(true);
        }
    }

    public static void enable(Node... nodes) {
        for (Node node : nodes) {
            node.setDisable(false);
        }
    }

    public static Button buildButton(String text) {
        Button button = new Button(text);
        button.setMnemonicParsing(false);
        button.setMinWidth(USE_PREF_SIZE);
        button.setMaxWidth(USE_PREF_SIZE);
        button.setPrefWidth(30);

        button.setMinHeight(USE_PREF_SIZE);
        button.setMaxHeight(USE_PREF_SIZE);
        button.setPrefHeight(25);
        return button;
    }

    public static <T extends Region> void setWidth(T node, Number value) {
        setWidth(node, value.doubleValue());
    }

    public static <T extends Region> void setWidth(T node, double value) {
        node.setMaxWidth(value);
        node.setMinWidth(value);
        node.setPrefWidth(value);
    }

    public static double computeTextContainerWidth(String text, Font font) {
        FontLoader fontLoader = Toolkit.getToolkit().getFontLoader();
        float width = fontLoader.computeStringWidth(text, font);
        return width * 1.3;
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/util/UIUtil.kt
package com.github.kindrat.cassandra.client.util

import javafx.scene.Node
import javafx.scene.control.Button
import javafx.scene.layout.AnchorPane
import javafx.scene.layout.Region
import javafx.scene.text.Font
import javafx.scene.text.Text
import org.apache.commons.lang3.StringUtils

object UIUtil {
    @JvmStatic
    fun fillParent(node: Node) {
        AnchorPane.setLeftAnchor(node, 0.0)
        AnchorPane.setRightAnchor(node, 0.0)
        AnchorPane.setTopAnchor(node, 0.0)
        AnchorPane.setBottomAnchor(node, 0.0)
    }

    @JvmStatic
    fun parseWords(rawString: String): Array<String> {
        return StringUtils.split(StringUtils.defaultString(rawString, ""))
    }

    @JvmStatic
    fun disable(vararg nodes: Node) {
        for (node in nodes) {
            node.isDisable = true
        }
    }

    @JvmStatic
    fun enable(vararg nodes: Node) {
        for (node in nodes) {
            node.isDisable = false
        }
    }

    @JvmStatic
    fun buildButton(text: String): Button {
        val button = Button(text)
        button.isMnemonicParsing = false
        button.minWidth = Region.USE_PREF_SIZE
        button.maxWidth = Region.USE_PREF_SIZE
        button.prefWidth = 30.0
        button.minHeight = Region.USE_PREF_SIZE
        button.maxHeight = Region.USE_PREF_SIZE
        button.prefHeight = 25.0
        return button
    }

    fun <T : Region?> setWidth(node: T, value: Number) {
        setWidth(node, value.toDouble())
    }

    @JvmStatic
    fun <T : Region?> setWidth(node: T, value: Double) {
        node!!.maxWidth = value
        node.minWidth = value
        node.prefWidth = value
    }

    @JvmStatic
    fun computeTextContainerWidth(text: String, font: Font): Double {
        val theText = Text(text)
        theText.font = font
        val width = theText.boundsInLocal.width
        return width * 1.3
    }
}